timemachine/fe/single_topology_v3.py

```diff
@@ -483,6 +484,221 @@ def find_dummy_groups_and_anchors(mol_a, mol_b, core_a, core_b):
+
+    lamb : float
+        alchemical parameter
+
+    k_min, lambda_min, lambda_max : float
+        see docstring of `interpolate_harmonic_force_constant` for documentation of these parameters
+    """
+
+    src_k, src_x = src_params
+    dst_k, dst_x = dst_params
+
+    k = interpolate_harmonic_force_constant(src_k, dst_k, lamb, k_min, lambda_min, lambda_max)
+    x = interpolate.linear_interpolation(src_x, dst_x, lamb)
+
+    return jnp.array([k, x])
+
+
+def cyclic_difference(a, b, period):
+    """
+    Returns the minimum difference between two points, with periodic boundaries.
+    I.e. the solution of ::
+
+        (a + x) % period = b % period
+
+    with minimum abs(x).
+    """
+
+    d = jnp.fmod(b - a, period)
+
+    def f(d):
+        return jnp.where(d <= period / 2, d, d - period)
+
+    return jnp.where(0 < d, f(d), -f(-d))
+
+
+def interpolate_harmonic_angle_params(src_params, dst_params, lamb, k_min, lambda_min, lambda_max):
+    """
+    Interpolate harmonic angle parameters using
+
+    1. Log-linear interpolation for force constants*
+    2. Shortest-path linear interpolation for equilibrium angles
+
+    * see note on special case when src_k=0 or dst_k=0 in the docstring of `interpolate_harmonic_force_constant`.
+
+    Parameters
+    ----------
+    src_params : array-like, float, (2,)
+        force constant and equilibrium angle at lambda=0
+
+    dst_params : array-like, float, (2,)
+        force constant and equilibrium angle at lambda=1
+
+    lamb : float
+        alchemical parameter
+
+    k_min, lambda_min, lambda_max : float
+        see docstring of `interpolate_harmonic_force_constant` for documentation of these parameters
+    """
+
+    src_k, src_phase = src_params
+    dst_k, dst_phase = dst_params
+
+    k = interpolate_harmonic_force_constant(src_k, dst_k, lamb, k_min, lambda_min, lambda_max)
+
+    phase = interpolate.linear_interpolation(
+        src_phase,
+        src_phase + cyclic_difference(src_phase, dst_phase, period=2 * np.pi),
+        lamb,
+    )
+
+    return jnp.array([k, phase])
+
+
+def interpolate_periodic_torsion_params(src_params, dst_params, lamb, lambda_min, lambda_max):
+    """
+    Interpolate periodic torsion parameters using
+
+    1. Linear interpolation for force constants (see note on ring
+       closing in `interpolate_harmonic_bond_params` docstring)
+    2. Linear interpolation for angles, using the shortest path
+    3. No interpolation for periodicity (pinned to source value)
+
+    * see note on special case when src_k=0 or dst_k=0 in the docstring of `interpolate_harmonic_force_constant`.
+
+    Parameters
+    ----------
+    src_params : array-like, float, (2,)
+        force constant, equilibrium dihedral angle, and periodicity at lambda=0
+
+    dst_params : array-like, float, (2,)
+        force constant and equilibrium dihedral angle, and periodicity at lambda=1
+
+    lamb : float
+        alchemical parameter
+
+    k_min, lambda_min, lambda_max : float
```

Comment:
There is no k_min in the signature.

Reply:
Fixed in [4acc511](https://github.com/proteneer/timemachine/pull/820/commits/4acc511b0414be6e9286a1e7c2c9b1149cb81e87).